src/common/evaluable.ts

```diff
@@ -60,10 +60,17 @@ export interface Evaluable {
    * Simplifies this Evaluable when possible.
    *
    * @param {Context} ctx context for the evaluation
-   * @param {string[]} ignoreKeys keys to be considered present even if their not present in the context
+   * @param {false | string[]} strictKeys keys to be considered present even if they are not present in the context
+   * @param {false | string[]} optionalKeys keys to be considered not present unless they are also in `strictKeys`;
+   *  when `strictKeys` is set to `false` and `optionalKeys` is an array, every key that is not in `optionalKeys`
+   *  is considered to be present and thus will be evaluated
    * @returns {Result | Evaluable} simplified value or itself
    */
-  simplify(ctx: Context, ignoreKeys: string[]): Result | Evaluable
+  simplify(
```

Comment:
Another small thing for consideration: https://github.com/briza-insurance/illogical/blob/e7405e310742e179ef0d4981669b4b4a55064e0b/src/evaluable.ts#L48-L49
- Should we have simplification / serialization options (object), rather that expanding the args?
- As we may be adding features for simplification/serialization it maybe easier to just expand the options object param.

Reply:
I like it be it would be a breaking change if we introduce it now.

To avoid the breaking change. We would need to do something like:

```
simplify(context: Context, optionsOrIgnoreKeys: string[] | SimplifyOptions): Evaluated | Evaluable
```

Also hiding this in options might be hard for the developer to realise he is introducing a breaking change. Changing a type of a property in the options will be a breaking change but is hard to see that.

I would not do it as part of this PR.

I believe that illogical needs a little bit of a revamp. There is the idea of getting rid of the classes and make it fully functional. We might want to push these kind of change to a newer major version of illogical. We could create this newer version without changing the evaluable specification, just change the implementation.